modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/auth/SecondaryStoreUserLoginTestCase.java

```diff
@@ -0,0 +1,259 @@
+import org.apache.http.impl.client.BasicCookieStore;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.message.BasicNameValuePair;
+import org.apache.http.util.EntityUtils;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import org.wso2.carbon.identity.application.common.model.xsd.ServiceProvider;
+import org.wso2.carbon.identity.user.store.configuration.stub.dto.UserStoreDTO;
+import org.wso2.identity.integration.common.clients.UserManagementClient;
+import org.wso2.identity.integration.common.clients.user.store.config.UserStoreConfigAdminServiceClient;
+import org.wso2.identity.integration.common.utils.UserStoreConfigUtils;
+import org.wso2.identity.integration.test.base.TomcatInitializerTestCase;
+import org.wso2.identity.integration.test.oidc.OIDCAbstractIntegrationTest;
+import org.wso2.identity.integration.test.oidc.OIDCUtilTest;
+import org.wso2.identity.integration.test.oidc.bean.OIDCApplication;
+import org.wso2.identity.integration.test.util.Utils;
+import org.wso2.identity.integration.test.utils.DataExtractUtil;
+import org.wso2.identity.integration.test.utils.OAuth2Constant;
+
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.wso2.identity.integration.test.utils.OAuth2Constant.AUTH_CODE_BODY_ELEMENT;
+
+/**
+ * This class contains test case for authentication of users in both primary and secondary user stores.
+ */
+
+public class SecondaryStoreUserLoginTestCase extends OIDCAbstractIntegrationTest {
+
+    private OIDCApplication playgroundApp;
+    protected HttpClient client;
+    protected List<NameValuePair> consentParameters = new ArrayList<>();
+    protected String sessionDataKeyConsent;
+    protected String sessionDataKey;
+    protected static final String PRIMARY_USERNAME = "primaryUsername";
+    protected static final String PRIMARY_PASSWORD = "primaryPassword";
+    protected static final String SECONDARY_USERNAME = "secondaryUsername";
+    protected static final String SECONDARY_PASSWORD = "secondaryPassword";
+    private static final String PRIMARY_USER_ROLE = "jdbcUserStoreRole";
+    private static final String SECONDARY_USER_ROLE = "WSO2TEST.COM/jdbcUserStoreRole";
+    private UserStoreConfigAdminServiceClient userStoreConfigAdminServiceClient;
+    private static final UserStoreConfigUtils userStoreConfigUtils = new UserStoreConfigUtils();
+    private UserManagementClient userMgtClient;
+    private static final String PERMISSION_LOGIN = "/permission/admin/login";
+    private static final String JDBC_CLASS = "org.wso2.carbon.user.core.jdbc.UniqueIDJDBCUserStoreManager";
+    private static final String DOMAIN_ID = "WSO2TEST.COM";
+    private static final String USER_STORE_DB_NAME = "SECONDARY_USER_STORE_DB";
```

Comment:
Not in the correct order. Please use the code styles.

Reply:
Done